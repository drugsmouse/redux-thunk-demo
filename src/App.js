import React, { Component } from "react";
import { connect } from "react-redux";

import fetchHeroes from "./actions/heroes";
import App from "./components/App";

class AppContainer extends Component {
  componentDidMount() {
    this.props.fetchHeroes();
  }

  render() {
    const { heroes, loading } = this.props;
    return <App heroes={heroes} loading={loading} />;
  }
}

const mapStateToProps = state => ({
  heroes: state.heroes.items,
  loading: state.heroes.loading
});

const mapDispatchToProps = dispatch => ({
  fetchHeroes: () => dispatch(fetchHeroes())
});

export default connect(mapStateToProps, mapDispatchToProps)(AppContainer);
