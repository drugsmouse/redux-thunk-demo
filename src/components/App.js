import React from "react";
import PropTypes from "prop-types";
import { Switch, Route } from "react-router-dom";

import logo from "../logo.svg";
import "../App.css";
import Navigation from "./Navigation";
import NotFound from "./NotFound";
import Home from "./Home";
import Hero from "./Hero";
import AddHero from "./AddHero";

const getHeroByName = (heroes, name) => heroes.find(hero => hero.name === name);

const App = ({ heroes, loading }) => (
  <div className="App">
    <header className="App-header">
      <img src={logo} className="App-logo" alt="logo" />
    </header>
    <Navigation heroes={heroes} />
    <AddHero />

    <Switch>
      <Route path={"/"} exact={true} component={Home} />
      <AddHero />
      {heroes.length && (
        <Route
          path={"/hero/:name"}
          render={({ match }) => {
            return <Hero hero={getHeroByName(heroes, match.params.name)} />;
          }}
        />
      )}
      <Route component={NotFound} />
    </Switch>
  </div>
);

App.propTypes = {
  heroes: PropTypes.array.isRequired,
  loading: PropTypes.bool.isRequired
};

export default App;
