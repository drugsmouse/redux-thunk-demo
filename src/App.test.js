import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('should render shallow', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<App user={true}/>);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot()
});

it('should fully render component', () => {
  const component = renderer.create(
    <App user={false}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

it('should test nested object', () => {

  function getObject () {
    return {
      name: 'Jack',
      position: {
        title: 'D3',
        salary: '10 00000 $'
      },
      years: '5'
    }
  }
  const actualResult = getObject();


  expect(actualResult).toMatchSnapshot()
})
