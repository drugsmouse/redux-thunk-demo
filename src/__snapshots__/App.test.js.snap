// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should fully render component 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to react
    </h1>
    invalid
  </header>
  <div>
    Some component throw 'asdf'  
  </div>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;

exports[`should render shallow 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to react
    </h1>
    valid
  </header>
  <Some />
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;

exports[`should test nested object 1`] = `
Object {
  "name": "Jack",
  "position": Object {
    "salary": "10 00000 $",
    "title": "D3",
  },
  "years": "5",
}
`;
